### PR TITLE
V2 view

### PR DESCRIPTION
**Pull request by Cool04ek**

### CODE REVIEW

File: lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/CoordinatorLayoutView.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import android.widget.FrameLayout;
+
+public class CoordinatorLayoutView extends CoordinatorLayout {
+
+	public CoordinatorLayoutView(Context context) {
+		super(context);
+		init();
+	}
+
+	public CoordinatorLayoutView(Context context, AttributeSet attrs) {
```
Comment (DanielZlotin) [inline lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/CoordinatorLayoutView.java]: remove additional unused constructors, and inline the `init` method to be inside constructor

File: lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/CoordinatorLayoutView.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+					MeasureSpec.makeMeasureSpec(getWidth(), MeasureSpec.EXACTLY),
+					MeasureSpec.makeMeasureSpec(getHeight(), MeasureSpec.EXACTLY));
+			layout(getLeft(), getTop(), getRight(), getBottom());
+		}
+	};
+
+	@Override
+	public void requestLayout() {
+		super.requestLayout();
+		post(measureAndLayout);
```
Comment (DanielZlotin) [inline lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/CoordinatorLayoutView.java]: why is this needed? you already set MATCH_PARENT on yourself

File: lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/FloatingActionButtonManager.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+			params.gravity = params.gravity | Gravity.RIGHT;
+			fab.setLayoutParams(params);
+		}
+	}
+
+	@ReactProp(name = "gravityLeft")
+	public void setGravityLeft(final FloatingActionButtonView fab, boolean gravityLeft) {
+		if (gravityLeft) {
+			CoordinatorLayout.LayoutParams params = (CoordinatorLayout.LayoutParams) fab.getLayoutParams();
+			if ((params.gravity & Gravity.RIGHT) == Gravity.RIGHT) {
```
Comment (DanielZlotin) [inline lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/FloatingActionButtonManager.java]: I don't know what these are, but I'm guessing they could be refactored out to a helper function?

Comment (DanielZlotin) [reply lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/FloatingActionButtonManager.java]: additionally, all this logic should probably be inside the view

File: lib/android/app/src/main/res/values/dimen.xml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="margin">16dp</dimen>
```
Comment (DanielZlotin) [inline lib/android/app/src/main/res/values/dimen.xml]: why do we need it in xml? move it to a constant where it's used

File: package.json [hunk 5]
```diff
@@ -45,7 +45,8 @@
     "react-native": "*"
   },
   "dependencies": {
-    "lodash": "4.x.x"
+    "lodash": "4.x.x",
+    "prop-types": "^15.6.0"
```
Comment (DanielZlotin) [inline package.json]: this should not be a dependency, it should go into devDependencies and peerDependencies: and with "version *"

File: playground/src/containers/LifecycleScreen.js [hunk 6]
```diff
@@ -3,7 +3,7 @@ const { Component } = require('react');
 
 const { View, Text, Button } = require('react-native');
 
-const Navigation = require('react-native-navigation');
+const { Navigation } = require('react-native-navigation');
```
Comment (DanielZlotin) [inline playground/src/containers/LifecycleScreen.js]: why the destructuring? this should stay a simple export

File: lib/src/index.js [hunk 7]
```diff
@@ -1,4 +1,12 @@
...
+export {
+  Navigation,
+  FloatingActionButton,
+  CoordinatorLayout
+};
+
+export default Navigation;
 
-module.exports = Navigation;
-module.exports.default = Navigation;
```
Comment (DanielZlotin) [inline lib/src/index.js]: undo this change. Navigation should be the only entry point. these 2 classes should be static properties of the internal Navigation class

File: lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/CoordinatorLayoutView.java (hunk 2 above)
Comment (Cool04ek) [reply lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/CoordinatorLayoutView.java]: It's some kind of bug from RN. It doesn't measure children properly without this 'hack'

File: lib/android/app/src/main/res/values/dimen.xml (hunk 4 above)
Comment (Cool04ek) [reply lib/android/app/src/main/res/values/dimen.xml]: if it's constant, you need to handle converting dp to px. It's just easier with .xml

File: playground/src/containers/LifecycleScreen.js (hunk 6 above)
Comment (Cool04ek) [reply playground/src/containers/LifecycleScreen.js]: it's because of exporting not only Navigation. It can be changed after improving issue with export

File: lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/CoordinatorLayoutView.java (hunk 2 above)
Comment (DanielZlotin) [reply lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/CoordinatorLayoutView.java]: hmm. it's worth a comment then, explaining that. maybe RN50 solves it?..

File: lib/android/app/src/main/res/values/dimen.xml (hunk 4 above)
Comment (DanielZlotin) [reply lib/android/app/src/main/res/values/dimen.xml]: I don't think it's a good idea to put such a general constant in the resources. which margin? where? also, how are the res being merged with RN apps? did you check?
